app/services/enricher.rb

```diff
@@ -0,0 +1,32 @@
+# Responsible for enriching a solr document and atomically updating the
+# document.
+class Enricher
+  pattr_initialize(:id)
+
+  # Perform enrichment
+  def enrich!
+    connection.update(
+      :params => { softCommit: true },
+      :data => [enriched_solr_document].to_json,
+      :headers => { 'Content-Type' => 'application/json' }
+    ) unless enriched_solr_document.empty?
```

Comment:
You're right! I copied the postfix from spotlight and totally ignored my own advice, good catch.